File: thesis/src/main/java/com/ptfunze/thesis/service/PostService.java
package com.ptfunze.thesis.service;

import com.ptfunze.thesis.dto.PostDto;
import com.ptfunze.thesis.request.CommentRequest;
import org.springframework.data.domain.Page;

import java.util.List;
import java.util.UUID;

public interface PostService {
    Page<PostDto> getAllPosts(int pageNo, int pageSize, String sortBy, String sortDir);
    PostDto createPost(PostDto postDto);
    PostDto getPostById(UUID id);
    void likePost(UUID id);
    void commentPost(CommentRequest commentRequest);
}

File: UI/app/services/PostService.js
import { ApiRequest } from "../../config";

export const addNewPost = async (post) => {
  const response = await ApiRequest().post(`posts`, post);
  return response.data;
};

export const getPosts = async (pageNo = 0) => {
  return (
    ApiRequest().get(
      `posts?pageNo=${pageNo}&pageSize=8&sortBy=createdOn&sortDir=desc`
    ) || []
  );
};

export const likePost = async (likeRequest) => {
  const response = await ApiRequest().post(`posts/like`, likeRequest);
  return response.data;
};

export const commentPost = async (commentRequest) => {
  const response = await ApiRequest().post(`posts/comment`, commentRequest);
  return response.data;
};
